### ISSUE TITLE
USP0008 is suppressing unrelated warnings

### ISSUE THREAD

**Issue by sailro** [microsoft/Microsoft.Unity.Analyzers#65]
#### Bug description

- Version of analyzers assembly: master
- Analyzer rule: `USP0008`

#### To Reproduce
```csharp
using UnityEngine;

public class Rotator : MonoBehaviour {

	void Update () 
	{
		Invoke("InvokeMe", 0f);
	}

	// this one is (and should be suppressed)
	void InvokeMe()
	{

	}
}
public class Unrelated : MonoBehaviour
{
	// we should have IDE0051 for this, but we suppress it
	void InvokeMe()
	{

	}
}
```

#### Expected behavior
In this case, we should not suppress `IDE0051` for `Unrelated.InvokeMe`.
We do that because of https://github.com/microsoft/Microsoft.Unity.Analyzers/blob/master/src/Microsoft.Unity.Analyzers/UnusedMethodSuppressor.cs#L54 which is completely wrong.
